### PR TITLE
feat(react): Route to react components; use react UISref components

### PR DESCRIPTION
**Pull request by christopherthielen**
- Removes ` @uirouter/angularjs`
- Adds `@uirouter/react-hybrid` which depends on `@uirouter/react` and `@uirouter/angularjs`
- Switch to `@uirouter/react` components `UISref` and `UISrefActive` and remove `inheritedData()` workaround
- Routes to `LoadBalancers` react component
- Bumps `core` and `amazon` versions

### CODE REVIEW

**Comment by christopherthielen** [comment]
@jrsquared @anotherchrisberry @icfantv PTAL

File: app/scripts/modules/amazon/src/loadBalancer/TargetGroup.tsx [hunk 1]
```diff
@@ -98,16 +32,29 @@ export class TargetGroup extends React.Component<ITargetGroupProps, ITargetGroup
...
-        <div className={`clickable clickable-row row no-margin-top target-group-header${this.state.active ? ' active' : ''}`} onClick={this.loadDetails}>
-          <div className="col-md-8 target-group-title">
-              {targetGroup.name}
-          </div>
-          <div className="col-md-4 text-right">
-            <HealthCounts container={targetGroup.instanceCounts}/>
-          </div>
-        </div>
+      <div className="target-group-container container-fluid no-padding">
+        <UISrefActive class="active">
```
Comment (christopherthielen) [inline app/scripts/modules/amazon/src/loadBalancer/TargetGroup.tsx]: This is how you use UISrefActive. The first dom element rendered (it's the `<div>` below) gets the class applied when the enclosed `UISref` is active.

File: app/scripts/modules/amazon/src/loadBalancer/TargetGroup.tsx [hunk 2]
```diff
@@ -98,16 +32,29 @@ export class TargetGroup extends React.Component<ITargetGroupProps, ITargetGroup
...
+          <UISref to=".targetGroupDetails" params={params}>
```
Comment (christopherthielen) [inline app/scripts/modules/amazon/src/loadBalancer/TargetGroup.tsx]: This is how you use `UISref`.  The first dom element rendered (again, the `<div>`) has a click handler added to it.

File: app/scripts/modules/core/src/loadBalancer/loadBalancer.states.ts [hunk 3]
```diff
@@ -69,7 +70,7 @@ module(LOAD_BALANCER_STATES, [
         template: '<load-balancer-filter app="$resolve.app"></load-balancer-filter>',
       },
       'master': {
-        template: '<load-balancers app="$resolve.app" style="display: flex;flex: 1 1 auto"></load-balancers>',
+        component: LoadBalancers, $type: 'react'
```
Comment (christopherthielen) [inline app/scripts/modules/core/src/loadBalancer/loadBalancer.states.ts]: This is how you route to a React Component.  The resolve data is bound as props (in this case, `props.app` receives the resolved `app` value).

File: app/scripts/modules/core/src/navigation/state.provider.ts [hunk 4]
```diff
@@ -1,14 +1,26 @@
...
 
 import { STATE_HELPER, StateHelper } from './stateHelper.provider';
 import { IFilterConfig } from '../filterModel/IFilterModel';
 
 import './navigation.less';
+import { ReactViewDeclaration } from '@uirouter/react';
 
-export interface INestedState extends Ng1StateDeclaration {
+// Typescript kludge to widen interfaces so INestedState can support both react and angular views
+export interface IReactHybridIntermediate extends Ng1StateDeclaration {
```
Comment (christopherthielen) [inline app/scripts/modules/core/src/navigation/state.provider.ts]: pay no attention to that code behind the curtain

File: app/scripts/modules/core/src/reactShims/react.uirouter.css [hunk 5]
```diff
@@ -0,0 +1,4 @@
+react-ui-view-adapter {
```
Comment (christopherthielen) [inline app/scripts/modules/core/src/reactShims/react.uirouter.css]: an extra custom element is created inside a `<ui-view>` which messes up the layout.  This fixes it (for load balancers, but we may need to tweak selectors going forward)

File: app/scripts/modules/core/src/reactShims/react.uirouter.tsx [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+import { StateObject, UIRouter } from '@uirouter/core';
+import './react.uirouter.css';
+
+export const REACT_UIROUTER = 'spinnaker.core.react.uirouter';
+const hybridModule = module(REACT_UIROUTER, ['ui.router']);
+
+// UI-Router React 0.5.0 provides resolve data as a prop called `resolves`.
+// This decorator spreads each individual resolved value to its _own prop_.
+//
+// e.g., instead of `props.resolves.myresolve` you can access `props.myresolve`
```
Comment (christopherthielen) [inline app/scripts/modules/core/src/reactShims/react.uirouter.tsx]: I'm working with the ui-router react guy to make this behavior the default

File: .gitignore [hunk 7]
```diff
@@ -20,6 +20,8 @@ test/google/node_modules/
 .awcache/
 .gradle/
 .happypack/
+.yalc
+yalc.lock
```
Comment (anotherchrisberry) [inline .gitignore]: oh you're doing it

**Comment by anotherchrisberry** [comment]
LGTM - will need to bump amazon/core versions after rebasing

File: app/scripts/modules/core/src/navigation/state.provider.ts (hunk 4 above)
Comment (jrsquared) [inline app/scripts/modules/core/src/navigation/state.provider.ts]: 💀

**Pull request merged**